internal/webhook/csv.go

```diff
@@ -0,0 +1,110 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-logr/logr"
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookPath = "/mutate-operators-coreos-com-v1alpha1-clusterserviceversion"
+)
+
+var (
+	ManagedPkgNames = []string{
+		// base csvs
+		"ocs-operator",
+		"rook-ceph-operator",
+		"ocs-client-operator",
+		"mcg-operator",
+		"noobaa-operator",
+		"odf-csi-addons-operator",
+		"csi-addons",
+		"cephcsi-operator",
+		"odf-prometheus-operator",
+		"recipe",
+
+		// extended csvs
+		"ibm-storage-odf-operator",
+	}
+)
+
+type ClusterServiceVersionDefaulter struct {
+	client.Client
+	Decoder           admission.Decoder
+	context           context.Context
+	logger            logr.Logger
+	OperatorNamespace string
+}
+
+// +kubebuilder:rbac:groups=operators.coreos.com,resources=clusterserviceversions,verbs=create;update;patch
+
+func (r *ClusterServiceVersionDefaulter) Handle(ctx context.Context, req admission.Request) admission.Response {
+	r.context = ctx
+	r.logger = log.FromContext(r.context)
+	r.logger.Info("Request received for ClusterServiceVersion review")
+
+	instance := &opv1a1.ClusterServiceVersion{}
```

Comment:
pls use the shortname, instance doesn't say anything. same in remaining places.
```suggestion
	csv := &opv1a1.ClusterServiceVersion{}
```